### PR TITLE
Conditional required roles

### PR DESCRIPTION
**Pull request by ZhengzheYang**
Bind roles only when the --enable-* flags are passed in by the user.  The test currently runs manually. Automation will come in separate PRs.

### CODE REVIEW

File: scripts/asm-installer/install_asm [hunk 1]
```diff
@@ -1803,46 +1804,70 @@ EOF
...
+    echo meshconfig.googleapis.com
+  fi
+  if can_modify_gcp_iam_roles; then
+    echo cloudresourcemanager.googleapis.com
+    echo iamcredentials.googleapis.com
+  fi
+  if can_register_cluster; then
+    echo gkeconnect.googleapis.com
+    echo gkehub.googleapis.com
+  fi
```
Comment (zerobfd) [inline scripts/asm-installer/install_asm:1862]: These APIs are required all of the time.

**Review by zerobfd — CHANGES_REQUESTED** [review]
Other than the one issue, the code looks good. We'll have to think a bit on how to test this properly.

File: scripts/asm-installer/install_asm (hunk 1 above)
Comment (ZhengzheYang) [reply scripts/asm-installer/install_asm]: Updated.

**Review by zerobfd — APPROVED** [review]
Seems fine to me. Couple of small comments. My one request would be to see if you can print out the permissions on the PermTest role when the test ends, regardless of whether it passes or fails.

File: scripts/asm-installer/tests/run_basic_suite_required_roles [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+cd "${SDIR}"
+
+main() {
+  # CLI setup
+  parse_args "$@"
+  local REVISION_LABEL; REVISION_LABEL="$(uniq_name "${SCRIPT_NAME}" "${BUILD_ID}")"
+  local EXPECTED_ROLES; EXPECTED_ROLES="projects/asm-scriptaro-oss/roles/PermTest
+roles/container.admin
+roles/iam.serviceAccountAdmin
+roles/resourcemanager.projectIamAdmin"
```
[the remark this answers is not in the document]
Comment (ZhengzheYang) [reply scripts/asm-installer/tests/run_basic_suite_required_roles]: Done

File: scripts/asm-installer/tests/common.sh [hunk 3]
```diff
@@ -922,3 +991,25 @@ EOF
     false
   fi
 }
+
+# From install_asm
```
[the remark this answers is not in the document]
Comment (ZhengzheYang) [reply scripts/asm-installer/tests/common.sh]: Done

**Pull request merged**